tauri-app/src-tauri/src/commands/order.rs

```diff
@@ -115,3 +116,52 @@ pub async fn order_remove(
 
     Ok(())
 }
+
+#[tauri::command]
+pub async fn order_add_calldata(
+    app_handle: AppHandle,
+    dotrain: String,
+    deployment: Deployment,
+    transaction_args: TransactionArgs,
+) -> CommandResult<Bytes> {
+    let add_order_args = AddOrderArgs::new_from_deployment(dotrain, deployment).await?;
+    let calldata = add_order_args.get_add_order_calldata(transaction_args)
```

Comment:
I think its a bit of a shame that we can't use the WritableTransaction from alloy-ethers-typecast to make the actual contract call -- as we loose the really nice transaction status notice implementation. That said it seems like it would be very complex to implement a walletconnect SingerMiddleware for ethers-rs, and is probably not worth it for now.